### ISSUE TITLE
Separate API from specific NPCs

### ISSUE THREAD

**Issue by hkzorman** [hkzorman/advanced_npc#52]
The proposal here is to make the mod `advanced_npc` completely as an API, with zero specific implementations of NPCs. Below is a comprehensive list of files that will be removed from the project and will be re-used in a separate, brand-new mod:

- textures
  - npc_child_female1.png
  - npc_child_male1.png
  - npc_female[1-11].png
  - npc_male[1-14].png
  - npc_male_priest
- random_data.lua
- data
  - occupations
    - default.lua
    - default_farmer.lua
    - default_miner.lua
    - default_priest.lua
  - dialogues_data.lua
  - gift_items_data.lua
  - names_data.lua

All these files will be re-used in another mod.

Also, minor changes in `npc.lua` are required as currently it depends on `random_data.lua` for names.
When this is implemented, `advanced_npc` will be a pure API for implementing NPCs.

**Comment by BrunoMine** [comment]
I believe this is much important because this textures can be completelly unseless in some situations.

**Comment by hkzorman** [comment]
Agreed, textures should be provided by the mods using `advanced_npc`, not the API itself. Thanks.
This is my next task to work on.
